xapian-applications/omega/Makefile.am

```diff
@@ -178,6 +178,10 @@ omindex_LDADD = $(MAGIC_LIBS) $(XAPIAN_LIBS) $(ZLIB_LIBS) libutf8convert.la
 omindex_pdf_SOURCES = assistant.cc worker_comms.cc common/str.cc handler_poppler.cc
 omindex_pdf_LDADD = $(POPPLER_LIBS) $(XAPIAN_LIBS)
 
+omindex_ebook_SOURCES = assistant.cc worker_comms.cc handler_libebook.cc
+omindex_ebook_LDADD = $(LIBEBOOK_LIBS) $(XAPIAN_LIBS)
+omindex_ebook_CPPFLAGS = $(AM_CPPFLAGS) $(LIBEBOOK_CFLAGS)
```

Comment:
This is wrong - `CPPFLAGS` are for the preprocessor, `CFLAGS` for the compiler.

This should be:

```
omindex_ebook_CXXFLAGS = $(LIBEBOOK_CFLAGS)
```

(We need to put them in CXXFLAGS so they're used for C++ compilation - this does kind of assume the C and C++ compilers are from the same toolchain, or at least compatible in the flags they take, but that's hard to avoid with `pkg-config`'s model of the world).

It looks like you're also missing the equivalent above:

```
omindex_pdf_CXXFLAGS = $(POPPLER_CFLAGS)
```